presto-geospatial/src/main/java/com/facebook/presto/geospatial/BingTileFunctions.java

```diff
@@ -0,0 +1,312 @@
+     */
+    private static boolean isBoundingBox(OGCGeometry geometry, Envelope envelope)
+    {
+        if (geometry instanceof OGCPoint) {
+            return true;
+        }
+
+        if (!(geometry instanceof OGCPolygon)) {
+            return false;
+        }
+
+        OGCPolygon polygon = (OGCPolygon) geometry;
+        if (polygon.numInteriorRing() > 0) {
+            return false;
+        }
+
+        final MultiVertexGeometry multiVertex = (MultiVertexGeometry) polygon.getEsriGeometry();
+        if (multiVertex.getPointCount() != 4) {
+            return false;
+        }
+
+        Set<Point> corners = new HashSet<>();
+        corners.add(new Point(envelope.getXMin(), envelope.getYMin()));
+        corners.add(new Point(envelope.getXMin(), envelope.getYMax()));
+        corners.add(new Point(envelope.getXMax(), envelope.getYMin()));
+        corners.add(new Point(envelope.getXMax(), envelope.getYMax()));
+
+        for (int i = 0; i < 4; i++) {
+            Point point = multiVertex.getPoint(i);
+            if (!corners.contains(point)) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    private static void checkZoomLevel(long zoomLevel)
+    {
+        checkCondition(zoomLevel > 0, "Zoom level must be > 0");
+        checkCondition(zoomLevel <= MAX_ZOOM_LEVEL, "Zoom level must be <= %d", MAX_ZOOM_LEVEL);
+    }
+
+    private static void checkXYCoordinate(long coordinate, long zoomLevel)
+    {
+        checkCondition(coordinate >= 0 && coordinate < (1 << zoomLevel), "XY coordinates for a Bing tile at zoom level %d must be within [0, %d) range", zoomLevel, 1 << zoomLevel);
+    }
+
+    private static void checkQuadKey(@SqlType(StandardTypes.VARCHAR) Slice quadkey)
+    {
+        checkCondition(quadkey.length() > 0, "QuadKey must not be empty string");
+        checkCondition(quadkey.length() <= MAX_ZOOM_LEVEL, "QuadKey must be %d characters or less", MAX_ZOOM_LEVEL);
+    }
+
+    private static boolean validLatitude(double latitude)
+    {
+        return latitude >= MIN_LATITUDE && latitude <= MAX_LATITUDE;
+    }
+
+    private static boolean validLongitude(double longitude)
+    {
+        return longitude >= MIN_LONGITUDE && longitude <= MAX_LONGITUDE;
+    }
+
+    private static void checkCondition(boolean condition, String formatString, Object... args)
+    {
+        if (!condition) {
+            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, format(formatString, args));
+        }
+    }
+
+    private static double clip(double n, double minValue, double maxValue)
+    {
+        return Math.min(Math.max(n, minValue), maxValue);
+    }
+
+    private static int mapSize(int zoomLevel)
+    {
+        return 256 << zoomLevel;
+    }
+
+    /**
+     * Copy-pasted from https://msdn.microsoft.com/en-us/library/bb259689.aspx
+     */
+    private static BingTile latLonToTile(double latitude, double longitude, int zoomLevel)
```

Comment:
Don't abbreviate. This is a good example of why we have this rule: the function above is `latLong` but this is `latLon`. Using full names avoids such inconsistency and confusion.